inc/classes/subscriber/preload/class-partial-preload-subscriber.php

```diff
@@ -134,6 +134,9 @@ public function preload_after_automatic_cache_purge( $deleted ) {
 				if ( strpos( $file_path, '#' ) ) {
 					// URL with query string.
 					$file_path = preg_replace( '/#/', '?', $file_path, 1 );
+				} else {
+					$permalink_structure = get_option( 'permalink_structure' );
+					$file_path           = rtrim( $file_path, '/' ) . ( '/' === substr( $permalink_structure, -1 ) ? '/' : '' );
```

Comment:
To improve upon @Tabrisrp's suggestion, I'd suggest the following:

1. Remove this line of code `$permalink_structure = get_option( 'permalink_structure' );`
2. Only do the concatenation when needed, instead of each time. 

Why? 

- The code will execute a few nanoseconds faster. Why?
    - It doesn't have to create the variable in memory in PHP's symtable.
    - It doesn't have to lookup the value in memory that's wired to the `$permalink_structure` variable.
    - It doesn't concatenate an empty string to the end of the `$file_path`.
- It's more readable and understandable.
    - The option tells us what it represents and what will be returned from `get_option`.
    - The conditional tells us: "Hey, if the permalink structure ends with a `/`, then let's add that to the end of the file path."

```php
$file_path = untrailingslashit( $file_path );
if ( '/' === substr(  get_option( 'permalink_structure' ), -1 ) {
    $file_path .= '/';
}
```